File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cs
public class Solution {
    public int FindMin(int[] nums) {
        int l = 0;
        int r = nums.Length - 1;
        int res = nums[0];

        while (l <= r) {
            if(nums[l] < nums[r]) {
                res = Math.Min(nums[l], res);
                break;
            }

            int p = (l + r)/2;
            res = Math.Min(nums[p], res);
            if(nums[p] >= nums[l]) {
                l = p + 1;
            }
            else {
                r = p - 1;
            }

        }

        return res;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        frequency = [[] for i in range(len(nums) + 1)]

        for i in nums:
            count[i] = 1 + count.get(i, 0)
        
        for key, value in count.items():
            frequency[value].append(key)
        
        res = []

        for i in range(len(frequency) - 1, 0, -1):
            for num in frequency[i]:
                res.append(num)
                if(len(res) == k):
                    return res

        